provider/terraform/custom_code.rb

```diff
@@ -98,11 +107,22 @@ def config_test
         ))
       end
 
+      def ignore_read_extras
+        @ignore_read_extra ||= []
+        ignore_read_extra
+      end
+
+      def substitute_test_paths(config)
+        config = config.gsub('path/to/private.key', 'test-fixtures/ssl_cert/test.key')
+        config.gsub('path/to/certificate.crt', 'test-fixtures/ssl_cert/test.crt')
+      end
+
       def validate
         super
         check_property :name, String
         check_property :primary_resource_id, String
         check_property :vars, Hash
+        check_property :ignore_read_extra, String
```

Comment:
wait hold up, isn't this an array, not a string? how does this make it through validation?

Reply:
Because validation didn't work 😮

It turns out using `check_property` with an array means that it doesn't validate the children. We need to use `check_property_list` which probably means we should never ever use `check_property` with an Array?

We also use `check_property` with `:import_format`. While the code will (eventually) fail if you use a non-string, that's only because we later try using string methods on it. I'll submit a separate PR fixing that, checking for other occurrences, and also probably try to ban `Array`s with `check_property`.